### PR TITLE
Add role: transmissionx

### PR DESCRIPTION
**Pull request by giosann**
This adds transmission, my personal favorite torrent client. Capable of handling 20k+ torrents without much trouble even in gui / desktop client.

**suggestions and testing welcome**

### CODE REVIEW

**Comment by owine** [comment]
Out of curiosity, why not just do the `tranxmissionx` role with the default name in settings being blank which would be a "default" role?

**Comment by giosann** [comment]
> Out of curiosity, why not just do the `tranxmissionx` role with the default name in settings being blank which would be a "default" role?

Well... I don't really have a reason, I just went with it... It's a good idea though.

**Comment by giosann** [comment]
Removed transmission

File: roles/transmissionx/tasks/template.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@
       - "/mnt:/mnt"
       - "/opt/transmission{{ rolename }}:/opt/transmission{{ rolename }}"
       - "/mnt/local/downloads/torrents/transmission{{ rolename }}:/data"
+      - "/mnt/local/downloads/torrents/transmission{{ rolename }}/watched:/watch"
```
Comment (giosann) [inline roles/transmissionx/tasks/template.yml:46]: The watched folder cannot be changed: https://github.com/linuxserver/docker-transmission/issues/100

**Comment by giosann** [comment]
I'd apprecciate if someone could test this with https://github.com/transmission-remote-gui/transgui, It does not work for me, the webui is fine...

**Comment by giosann** [comment]
It works fine for me, I've tried it on a somewhat new Cloudbox install and on a completely fresh one.

There is a minor issue but I don't think it's solvable: When the container starts even if the settings.json file has the correct configuration the container will automatically create the /downloads/complete and /downloads/incomplete directories.

I've tested mapping /downloads/ to /mnt/local/downloads/torrents/transmission{role}/ but this creates issues with sonarr because the path mapping does not match.

In my opinion this is minor because it's just an empty folder. I'll note this in the wiki but I think this is mergeable now.

**Comment by saltydk** [comment]
This PR seems like a bit of a mess. Why is the opt folder mounted as 
```
- "/opt/transmission{{ rolename }}:/opt/transmission{{ rolename }}"
```
And you mount the download folders twice?

**Comment by giosann** [comment]
Well I'm glad it is a bit of a mess and not a complete mess hahaha.

opt is mounted like that because I copied from https://github.com/Cloudbox/Community/blob/develop/roles/transmissionvpn/tasks/main.yml and used the "if it ain't broke, don't fix it" rule.

as for this double mount

```
      - "/mnt/local/downloads/:/downloads"
      - "/mnt/local/downloads/torrents/transmission{{ rolename }}/watch:/watch"
```

It is because the container automatically changes the watched folder at each start and sets it to /watch so this is the only solution. See: https://github.com/linuxserver/docker-transmission/issues/100

**Comment by saltydk** [comment]
By double mount I meant 
```
volumes: "{{ default_volumes + torrents_downloads_path|default([]) }}"
```
You define the download mount yet that is already included by the default setup.

**Comment by saltydk** [comment]
As far as copying the vpn version that is not a good way of doing things when you're not using the same image.

**Review by saltydk — COMMENTED** [review]
This would be my suggestion for the role. Take it for a spin?

**Comment by giosann** [comment]
Everything seems to be working fine, both the already in use updated roles and the new ones I've put for testing

**Pull request merged**